app/Services/RequestLogService.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\UserRequestLog;
+use App\Models\UserSettings;
+use Closure;
+use Illuminate\Support\Facades\Auth;
+
+class RequestLogService
+{
+
+    public function collectData($request,$response){
+
+        UserRequestLog::create([
+            'user_id' => Auth::user()->id,
+            'url' => $request->url(),
+            'method' => $request->method(),
+            'passed' => '1',
+            'request_body' => $request,
+            'response_body' => $response,
+            'headers' => $request->header()
+        ]);
+
+        return response([
+            'message' => 'data collected successfully',
+        ], 200);
+    }
+
+    public static function checkUser(){
+
+        $user = Auth::user();
+
+        $numOfRequestsToday = $user->log->whereDate('created_at',today())->where('passed',1)->count();
```

Comment:
Research Laravel scopes;
add 2 scopes to the model: 
whereCreatedToday()
and
whereIsPassed()

and replace exiting ->where clauses with new scopes;